bookkeeper-common/src/main/java/org/apache/bookkeeper/common/component/ComponentStarter.java

```diff
@@ -72,6 +73,8 @@ public void run() {
 
         // register a component exception handler
         component.setExceptionHandler((t, e) -> {
+            log.error("Triggered exceptionHandler of Component: {} because of Exception in Thread: {}",
```

Comment:
Since we hook up exception handler on every component, in case if multiple of them fail, we might see IllegalStateException when we try to start a thread which is already started.

Reply:
@reddycharan Even if we share the same reference across all components, multiple of them can fail at the same time, causing the hook to trigger (even though it isn't correctly implemented yet). I am fine with addressing this later since this is not very likely to happen.

Reply:
yes, as we deiscussed it would be shared instance.

> multiple of them can fail at the same time, causing the hook to trigger
All we should care about is 'atleast-once' semantics. As long as 'shutdownHookThread' is executed once we should be fine. Anyhow in `LifecycleComponentStack.start` Components are started sequentially, so multiple of them failing at the same time is not going to happen.

Reply:
> All we should care about is 'atleast-once' semantics. As long as 'shutdownHookThread' is executed once we should be fine.

Yes this is not a major concern, all I was saying was that it would throw a bunch of `IllegalThreadStateException` at the end of logs, which can be potentially confusing.

> Anyhow in LifecycleComponentStack.start Components are started sequentially, so multiple of them failing at the same time is not going to happen.

It's not just during the start. Since we are passing the handler to the component, component may trigger it anytime during its lifecycle. It's OK to be not overthink much about it for now.